actions/docs-verifier/src/ActionRunner/Program.cs

```diff
@@ -42,16 +43,17 @@ static bool IsInWhatsNewDirectory(PullRequestFile file)
 // Also, filter out files that are part of the "What's new" directory - as they shouldn't require redirects.
 foreach (PullRequestFile file in files.Where(f => IsYmlOrMarkdownFile(f) && !IsInWhatsNewDirectory(f)))
 {
+    Matcher? matcher = DocfxConfigurationReader.GetMatcher();
     // Changing the extension from .yml to .md or the opposite doesn't require a redirection.
     // In both cases, the URL in live docs site is the same.
-    if (file.IsRenamed() && !IsExtensionChangeOnly(file.PreviousFileName, file.FileName))
+    if (file.IsRenamed() && matcher?.Match(file.PreviousFileName)?.HasMatches == true && !IsExtensionChangeOnly(file.PreviousFileName, file.FileName))
     {
         if (!await RedirectionsVerifier.WriteResultsAsync(Console.Out, file.PreviousFileName))
         {
             returnCode++;
         }
     }
-    else if (file.IsRemoved() && !files.Any(f => f.IsAdded() && IsExtensionChangeOnly(file.FileName, f.FileName)))
+    else if (file.IsRemoved() && matcher?.Match(file.FileName)?.HasMatches == true  && !files.Any(f => f.IsAdded() && IsExtensionChangeOnly(file.FileName, f.FileName)))
```

Comment:
Currently the code `matcher?.Match(......)` is duplicated, but I have another PR that introduced a local function `IsRedirectableFile`. which will avoid duplication (depends on which PR gets merged first, this will be fixed when conflicts in the other PR are fixed).